projects/laji/src/app/shared/service/user.service.ts

```diff
@@ -25,96 +25,113 @@ import { PlatformService } from '../../root/platform.service';
 import { BrowserService } from './browser.service';
 import { retryWithBackoff } from '../observable/operators/retry-with-backoff';
 import { httpOkError } from '../observable/operators/http-ok-error';
-import { PERSON_TOKEN } from './laji-api-worker-common';
 import { Profile } from '../model/Profile';
 import { Global } from '../../../environments/global';
 
-export interface ISettingResultList {
+export interface UserSettingsResultList {
   aggregateBy?: string[];
   selected?: string[];
   pageSize?: number;
 }
 
-export interface IUserSettings {
-  resultList?: string[];
-  taxonomyList?: string[];
+export interface UserSettings {
+  resultList?: UserSettingsResultList;
+  taxonomyList?: string[] | { selected: string[] };
   observationMap?: any;
   frontMap?: any;
   formDefault?: any;
   '_global_form_settings_'?: any; // eslint-disable-line @typescript-eslint/naming-convention
   [key: string]: any;
 }
 
-interface IPersistentState {
-  isLoggedIn: boolean;
+interface PersistentState {
+  token: string;
 }
 
-export interface IUserServiceState extends IPersistentState {
-  token: string;
+export interface UserServiceState extends PersistentState {
   user: Person | null;
-  settings: IUserSettings;
+  settings: UserSettings;
   allUsers: {[id: string]: Person|Observable<Person>};
 }
 
-const _persistentState: IPersistentState = {
-  isLoggedIn: false
+export const createProfile = (profile: Profile | null, user: Person | null): Profile => {
```

Comment:
I think `prepareProfile` was a better name. I associate the word "create" with actually creating some external entity, whereas "prepare" I associate with modifying some entity for internal usage. Could also be just simply "profile()", since it's about "serializing" it into our internal model of the profile.